### PR TITLE
Feature/travis

### LINKED ISSUE #25: Automated tests for the login page
Probably using [selenium](https://www.seleniumhq.org/).

Simulate user activities like mouse move over, clicking, and typing.

### PR DESCRIPTION
**Pull request by lyuboxin**
Resolved #25 implement automated tests for login page by travis

### CODE REVIEW

**Review by zhangjiangqige — DISMISSED** [review]

File: test.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+set -e
+export PATH=`pwd`/driver:$PATH
+
+echo $PATH
+
+ls driver
+
+python test.py
```
Comment (xingweitian) [inline test.sh]: add a new line at the end of the file

File: test.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+testLogin('unexist username','lyuboxin1@gmail','','username or password is wrong')
+
+
+
+if CLOASE_AFTER_TEST:
+    browser.quit()
+
+
+
+
```
Comment (xingweitian) [inline test.py]: make this file meets PEP8

File: test.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+__testUrl = 'http://ubuntu1604-006.student.cs.uwaterloo.ca:19487/'
+
+browser = Browser('chrome',headless=True)
+browser.visit(__testUrl)
+
+output("test page:"+ browser.title)
+
+
+
+testLogin("no username",'','','please enter your email')
```
Comment (xingweitian) [inline test.py]: choice one of them, ' or "

File: test.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import sys
+import re
+from splinter.browser import Browser
+
+CLOASE_AFTER_TEST =False
+GBK = "gbk"
```
Comment (xingweitian) [inline test.py]: we don't need to indicate gbk or utf8

File: test.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+encoding = lambda x:x.encode('gbk')
+
+def output(x):
+    print encoding(x)
+
+def resultMsg(x):
+        assert x==True
+        print 'pass'
+    # print '---------------------'
```
Comment (xingweitian) [inline test.py]: remove this line

**Review by xingweitian — CHANGES_REQUESTED** [review]

File: test.py [hunk 6]
```diff
@@ -0,0 +1,44 @@
+import sys
+import re
+from splinter.browser import Browser
+
+CLOASE_AFTER_TEST =False
```
Comment (xingweitian) [inline test.py]: please meet PEP8

**Review by xingweitian — CHANGES_REQUESTED** [review]

File: test.py [hunk 7]
```diff
@@ -23,22 +20,14 @@ def checkresult(x):
...
 
-output('test page:'+ browser.title)
-
-
-
-testLogin('no username','','','please enter your email')
-testLogin('no password','lyuboxin@gmail','','please enter your password')
-testLogin('unexist username','lyuboxin1@gmail','','username or password is wrong')
-
+output('test page:' + browser.title)
```
Comment (xingweitian) [inline test.py]: use print, not output

File: test.py [hunk 8]
```diff
@@ -23,22 +20,14 @@ def checkresult(x):
...
+testLogin('no password', 'lyuboxin@gmail', '', 'please enter your password')
+testLogin('unexist username', 'lyuboxin1@gmail', '', 'username or password is wrong')
 
 if CLOASE_AFTER_TEST:
-    browser.quit()
-
-
-
-
+    browser.quit()
```
Comment (xingweitian) [inline test.py]: add a new line at the end of the file

**Review by xingweitian — CHANGES_REQUESTED** [review]

**Review by zhangjiangqige — APPROVED** [review]
While it works for now, we need to be careful that sometimes the page need to be refreshed so the fields filled in previous test cases will not affect later cases.

**Review by xingweitian — APPROVED** [review]

**Pull request merged**